src/ui/public/query_manager/query_manager.js

```diff
@@ -1,79 +1,24 @@
 import _ from 'lodash';
-import { FilterManagerProvider } from 'ui/filter_manager';
-import { FilterBarLibMapAndFlattenFiltersProvider } from 'ui/filter_bar/lib/map_and_flatten_filters';
-import { FilterBarLibExtractTimeFilterProvider } from 'ui/filter_bar/lib/extract_time_filter';
-import { FilterBarLibChangeTimeFilterProvider } from 'ui/filter_bar/lib/change_time_filter';
-import { toKueryExpression, fromKueryExpression, nodeTypes, filterToKueryAST } from 'ui/kuery';
 
-export function QueryManagerProvider(Private) {
-  const filterManager = Private(FilterManagerProvider);
-  const mapAndFlattenFilters = Private(FilterBarLibMapAndFlattenFiltersProvider);
-  const extractTimeFilter = Private(FilterBarLibExtractTimeFilterProvider);
-  const changeTimeFilter = Private(FilterBarLibChangeTimeFilterProvider);
+export function queryManagerFactory(state) {
 
-  return function (state) {
-
-    function add(field, values = [], operation, index) {
```

Comment:
Once I exposed query manager in the Vis API I realized the `add` and `addLegacyFilter` functions really didn't belong in a generic query API. `add` is highly specific to the doc table with semantics that wouldn't make sense for other clients. `addLegacyFilter` was only useful for converting filters created by the filter_bar_click_handler. So I moved these functions to locations that made more sense to me. `add` went to the doc_table module, `addLegacyFilter` went to the filter_bar module.